src/migration/src/index_methods.rs

```diff
@@ -0,0 +1,46 @@
+use std::future::Future;
+
+use crate::{
+    canister_methods::Canister,
+    utils::{context, Environment, AGENT},
+};
+use candid::{CandidType, Principal};
+use catalyze_shared::{group::Group, profile::Profile};
+use eyre::Error;
+
+pub trait IndexCallsTrait<K: CandidType, V: CandidType> {
+    fn insert(k: K, v: V) -> impl Future<Output = Result<Vec<u8>, Error>> + Send;
+}
+
+pub struct ProfileIndexCalls;
+pub struct GroupIndexCalls;
+
+impl IndexCallsTrait<Principal, Profile> for ProfileIndexCalls {
+    async fn insert(k: Principal, v: Profile) -> Result<Vec<u8>, Error> {
+        context().indexes.profiles.update("insert", (k, v)).await
```

Comment:
why not just have method which returns right canister from the context, and default impl for the insert method?